### PR TITLE
EASYOPAC-514 - Adding ding_entity_buttons to teaser view mode.

### PR DESCRIPTION
**Pull request by bozeak**

### CODE REVIEW

File: modules/ding_availability/js/ding_availability.js [hunk 1]
```diff
@@ -92,6 +92,9 @@
     $.each(entity_ids, function(index, entity_id) {
       // Reserve button
       var reserver_btn = element.parents('.ting-object:first, .material-item:first').find('a[id$=' + entity_id + '].reserve-button');
+      if (!reserver_btn.length) {
+        reserver_btn = $('.ting-object:first').find('a[id$=' + entity_id + '].reserve-button');
```
Comment (guddo) [inline modules/ding_availability/js/ding_availability.js]: Can you tell why this is needed?

File: modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc [hunk 2]
```diff
@@ -18,6 +18,12 @@ function ding_ting_frontend_field_default_field_instances() {
     'description' => '',
     'ding_entity_type' => 'ding_entity_collection',
     'display' => array(
+      'collection_popup' => array(
```
Comment (guddo) [inline modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc]: How is `collection_popup` related to this change?

File: modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc [hunk 3]
```diff
@@ -160,17 +178,23 @@ function ding_ting_frontend_field_default_field_instances() {
         'type' => 'ding_availability_default',
         'weight' => 2,
       ),
+      'objects_in_popup' => array(
```
Comment (guddo) [inline modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc]: How is `objects_in_popup` related to this change?

File: modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc [hunk 4]
```diff
@@ -160,17 +178,23 @@ function ding_ting_frontend_field_default_field_instances() {
...
         'type' => 'hidden',
         'weight' => 33,
       ),
       'teaser' => array(
-        'label' => 'above',
+        'label' => 'hidden',
         'settings' => array(),
         'type' => 'hidden',
-        'weight' => 39,
+        'weight' => 2,
```
Comment (guddo) [inline modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc]: Why is the weight changed?

File: themes/ddbasic/templates/fields/field--ding-entity-rating-action--ting-object.tpl.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ * - $classes_array: Array of html class attribute values. It is flattened
+ *   into a string within the variable $classes.
+ *
+ * @see template_preprocess_field()
+ * @see
+ */
+?>
+
+<?php foreach ($items as $delta => $item): ?>
+  <div class="<?php print $classes; ?>"><?php print render($item); ?></div>
```
Comment (guddo) [inline themes/ddbasic/templates/fields/field--ding-entity-rating-action--ting-object.tpl.php]: Great 👍  But what about the Entity buttons? They are also wrapped in `.field-items.field-item`.

File: modules/ding_availability/js/ding_availability.js (hunk 1 above)
Comment (bozeak) [inline modules/ding_availability/js/ding_availability.js]: 'element' variable is forming of "Availability label" which is present on ting_object's default and 
search_result view modes, for teaser we don't have it I was forced to find a reservation link from present ting_object and to form new "reserver_btn" variable.

File: modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc (hunk 2 above)
Comment (bozeak) [inline modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc]: Will revert this.

File: modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc (hunk 3 above)
Comment (bozeak) [inline modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc]: Will revert this.

File: themes/ddbasic/templates/fields/field--ding-entity-rating-action--ting-object.tpl.php (hunk 5 above)
Comment (bozeak) [inline themes/ddbasic/templates/fields/field--ding-entity-rating-action--ting-object.tpl.php]: Will add one template and for entity buttons.

File: modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc (hunk 4 above)
Comment (bozeak) [inline modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc]: This must be a mistake, we don't use ding_availability_holdings. Will revert.

File: modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc [hunk 6]
```diff
@@ -241,16 +241,18 @@ function ding_ting_frontend_field_default_field_instances() {
         'weight' => 11,
       ),
       'search_result' => array(
-        'label' => 'above',
+        'label' => 'hidden',
```
Comment (guddo) [inline modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc]: This change is not related as well. This is because you have another module enabled.

File: modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc [hunk 7]
```diff
@@ -329,7 +331,7 @@ function ding_ting_frontend_field_default_field_instances() {
         'label' => 'above',
         'settings' => array(),
         'type' => 'hidden',
-        'weight' => 16,
+        'weight' => 31,
```
Comment (guddo) [inline modules/ding_ting_frontend/ding_ting_frontend.features.field_instance.inc]: Why the weight changes?

File: modules/ding_ting_frontend/ding_ting_frontend.field_group.inc [hunk 8]
```diff
@@ -375,14 +375,13 @@ function ding_ting_frontend_field_group_info() {
...
     'children' => array(
-      0 => 'ting_title',
-      1 => 'ting_series',
-      2 => 'ting_abstract',
-      3 => 'ting_author',
-      4 => 'ting_subjects',
-      5 => 'ding_serendipity_info',
-      6 => 'ding_entity_rating_action',
-      7 => 'ding_entity_rating_result',
+      0 => 'ding_entity_rating_action',
```
Comment (guddo) [inline modules/ding_ting_frontend/ding_ting_frontend.field_group.inc]: Why change order?
